### PR TITLE
Package the JDK into jdk.app on macOS

### PR DESCRIPTION
**Pull request by jasontedor**
This commit packages the bundled JDK into jdk.app on macOS to assist with notarization there.

### CODE REVIEW

**Comment by jasontedor** [comment]
@elasticmachine update branch

File: distribution/archives/build.gradle [hunk 1]
```diff
@@ -59,8 +59,14 @@ CopySpec archiveFiles(CopySpec modulesFiles, String distributionType, String pla
         with binFiles(distributionType, oss, jdk)
       }
       if (jdk) {
-        into('jdk') {
-          with jdkFiles(project, platform)
+        if ("darwin".equals(platform)) {
```
Comment (mark-vieira) [inline distribution/archives/build.gradle]: This could just be `into("darwin".equals(platform) ? 'jdk.app' : 'jdk')`.

**Review by mark-vieira — APPROVED** [review]
LGTM.

@jasontedor could you provide some context as to _how_ this helps with notarization? It's not immediately obvious to me what exactly renaming the directory we put the jdk in accomplishes to that effect.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by jasontedor** [comment]
@mark-vieira It puts the directory structure into the format that Apple expects for application bundles (`${name}.app/Contents/MacOS`). We're then signing and notarizing the `jdk.app` bundle the `controller.app` bundle (for machine learning binaries) separately.

File: distribution/archives/build.gradle (hunk 1 above)
Comment (jasontedor) [reply distribution/archives/build.gradle]: I pushed 5b2d681ac5178f7cd93830d9c80112c0087471da directly to master and 90d44375136e0b66608be59949ab2bad4710111a to 7.x.

**Pull request merged**